### ISSUE TITLE
[BUG] Main thread panic - byte index out of bounds

### ISSUE THREAD

**Issue by oraqlle** [Champii/Rock#86]
When running any program, the thread panics do to an out of bounds error. The output for the factorial example from the [README](https://github.com/Champii/Rock/blob/master/README.md/#quickstart) shows this error:

```zsh
$ rock run
thread 'main' panicked at 'byte index 20 is out of bounds of `main = print fact 4`', library/core/src/str/mod.rs:107:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

I might be able to fix it though.

**Comment by oraqlle** [comment]
After some testing, it seems that your parser is accessing past the buffer allocated storage causing a panic. Whatever buffer stores the `main` function from the `*.rk` file. Which part of the code looks for the main function in a rock source file?

**Comment by oraqlle** [comment]
If you put an empty line under the main function, the error disappears. However, no matter the case, multiline mains are still having issues.

**Comment by Champii** [comment]
It seems that the parser requires a `\n` at the end of each and every statements.
This is difficult to fix properly for now, as it would be weird to handle special cases for that.

One of the possible 'temporary' fix would be to add a `\n` at the end of every modules right before parsing.
The parser should properly handle the extra newlines if any

**Comment by Champii** [comment]
Fixed in #87
